core/config/capabilities_config.go

```diff
@@ -0,0 +1,11 @@
+package config
+
+type Capabilities interface {
+	Registry() CapabilityRegistry
+	Peering() P2P
+}
+
+type CapabilityRegistry interface {
+	Enabled() bool
```

Comment:
this is suspicious. do you really need an interface only to tell you if a feature is on or off? instead can you write a help func that infers it from whether or not something concrete is nil?

somewhere later in the code you are using if registry != nil, which seems to mix level of abstractions

Reply:
removing for now, as discussed offline.